client/blocks/upsell-nudge/index.jsx

```diff
@@ -0,0 +1,130 @@
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import classnames from 'classnames';
+import { localize } from 'i18n-calypso';
+import Gridicon from 'components/gridicon';
+
+/**
+ * Internal dependencies
+ */
+import { Button } from '@automattic/components';
+import TrackComponentView from 'lib/analytics/track-component-view';
+import { recordTracksEvent } from 'state/analytics/actions';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+export class UpsellNudge extends Component {
+	static defaultProps = {
+		className: '',
+	};
+
+	static propTypes = {
+		className: PropTypes.string,
+		eventName: PropTypes.string,
+		eventProperties: PropTypes.object,
+		buttonText: PropTypes.string,
+		icon: PropTypes.string,
+		href: PropTypes.string,
+		text: PropTypes.string,
+		onClick: PropTypes.func,
+		onDismissClick: PropTypes.func,
+		showDismiss: PropTypes.bool,
+		isCompact: PropTypes.bool,
+		track: PropTypes.func.isRequired,
+		translate: PropTypes.func.isRequired,
+	};
+
+	onClick = e => {
+		const { eventName, eventProperties, track, onClick } = this.props;
+		track( 'calypso_upsell_nudge_button_click', { event_name: eventName, ...eventProperties } );
+		if ( onClick ) {
+			onClick( e );
+		}
+	};
+
+	onDismissClick = e => {
+		const { eventName, eventProperties, track, onDismissClick } = this.props;
+		track( 'calypso_upsell_nudge_button_dismiss_click', {
+			event_name: eventName,
+			...eventProperties,
+		} );
+		if ( onDismissClick ) {
+			onDismissClick( e );
```

Comment:
Good question. I think the most common one would be a Tracks or analytics event on dismiss. But I think if we use the `Banner` component this becomes a non-issue, since it handles dismiss events already, right?

Reply:
>  But I think if we use the Banner component this becomes a non-issue, since it handles dismiss events already, right?

It uses dismissible card, but I'd be able to help with the wiring for a more compact view, if a card doesn't fit.